activerecord/lib/active_record/enum.rb

```diff
@@ -199,7 +199,14 @@ def _enum(name, values, prefix: nil, suffix: nil, scopes: true, **options)
             suffix == true ? "_#{name}" : "_#{suffix}"
           end
 
-          pairs = values.respond_to?(:each_pair) ? values.each_pair : values.each_with_index
+          pairs = if values.respond_to?(:each_pair)
+            values.each_pair
+          elsif column_backed_by_string?(name)
```

Comment:
Maybe this method knows too much about this feature? An alternative would be

```suggestion
          elsif column_type?(name) == :string
```